app/models/katello/ping.rb

```diff
@@ -5,20 +5,22 @@ class Ping
     PACKAGES = %w(katello candlepin pulp qpid foreman tfm hammer).freeze
 
     class << self
+      def pulpcore_enabled # for downstream 6.9, remove in 6.10
+        system('systemctl is-enabled pulpcore-api.service')
+      end
```

Comment:
lets add one more check here, Only check systemctl if these SETTINGS are set:

```
  :use_pulp_2_for_content_type:
    :docker: false
    :file: false
    :yum: false
```

so it'd be like:
```
if SETTING[:katello][:use_pulp_2_for_content_type] && SETTING[:katello][:use_pulp_2_for_content_type][:yum] && SETTING[:katello][:use_pulp_2_for_content_type][:docker] && SETTING[:katello][:use_pulp_2_for_content_type][:file]]  && system('systemctl is-enabled pulpcore-api.service')
```

This will 'narrow' down the scope of when we actually have to go to systemctl

Reply:
we want to check when these are true, updated my snippet to show that :)

Reply:
Thanks for the snippet :) Updated with that

Reply:
oh shoot, we need to check the opposite, sorry for leading you wrong, so we actually want:
```
SETTING[:katello][:use_pulp_2_for_content_type].nil? || 
( !SETTING[:katello][:use_pulp_2_for_content_type][:yum] &&
  !SETTING[:katello][:use_pulp_2_for_content_type][:docker] && 
  !SETTING[:katello][:use_pulp_2_for_content_type][:file]) || 
system('systemctl is-enabled pulpcore-api.service')
```

so the logic being either:
1) use_pulp_2_for_content_type is not set at all 
OR
2) use_pulp_2_for_content_type is set to false for all types
OR
3) systemctl is-enabled pulpcore-api.service   returns enabled


Couple of other things:
* the code as you have it just does an if check, doesn't actually return anything :)  i think you can drop the if and just return the output of the large expression

Reply:
Updated, thanks for the assist and catch on the if statement :)

Reply:
This needs to be SETTINGS, not SETTING

Don't ever trust pseudo code without trying it yourself ;)